aptible/resource_database.go

```diff
@@ -88,68 +94,140 @@ func resourceDatabase() *schema.Resource {
 
-	_ = d.Set("database_id", database.ID)
+	payload := aptibleapi.NewCreateOperationRequest("provision")
+	if containerSize != 0 {
+		payload.SetContainerSize(containerSize)
+	}
+	if iops != 0 {
+		payload.SetProvisionedIops(iops)
+	}
+	if profile != "" {
+		payload.SetInstanceProfile(profile)
+	}
+
+	op, _, err := client.
+		OperationsAPI.
+		CreateOperationForDatabase(ctx, db.Id).
+		CreateOperationRequest(*payload).
+		Execute()
+	if err != nil {
+		return err
+	}
 
-	d.SetId(strconv.Itoa(int(database.ID)))
+	del, err := legacy.WaitForOperation(int64(op.Id))
+	if err != nil {
+		return err
+	}
+	if del {
+		return fmt.Errorf("the replica with handle: %s was unexpectedly deleted", handle)
+	}
+
+	_ = d.Set("database_id", db.Id)
+
+	d.SetId(strconv.Itoa(int(db.Id)))
 	return resourceDatabaseRead(d, meta)
 }
 
 // syncs Terraform state with changes made via the API outside of Terraform
 func resourceDatabaseRead(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*providerMetadata).LegacyClient
-	databaseID := int64(d.Get("database_id").(int))
+	client := meta.(*providerMetadata).Client
+	databaseID := int32(d.Get("database_id").(int))
+	ctx := context.Background()
+	ctx = meta.(*providerMetadata).APIContext(ctx)
 
-	database, err := client.GetDatabase(databaseID)
+	database, resp, err := client.DatabasesAPI.GetDatabase(ctx, databaseID).Execute()
 	if err != nil {
-		log.Println(err)
 		return generateErrorFromClientError(err)
 	}
-
-	if database.Deleted {
+	if resp.StatusCode == http.StatusNotFound {
 		d.SetId("")
 		log.Println("Database with ID: " + strconv.Itoa(int(databaseID)) + " was deleted outside of Terraform. Now removing it from Terraform state.")
 		return nil
 	}
 
-	_ = d.Set("container_size", database.ContainerSize)
-	_ = d.Set("container_profile", database.ContainerProfile)
-	_ = d.Set("disk_size", database.DiskSize)
-	_ = d.Set("default_connection_url", database.DefaultConnection)
-	_ = d.Set("connection_urls", database.ConnectionURLs)
-	_ = d.Set("handle", database.Handle)
-	_ = d.Set("env_id", database.EnvironmentID)
-	_ = d.Set("database_type", database.Type)
-	_ = d.Set("database_image_id", database.DatabaseImage.ID)
-	_ = d.Set("version", database.DatabaseImage.Version)
+	urls := []string{}
+	creds := database.Embedded.GetDatabaseCredentials()
+	for _, cred := range creds {
+		urls = append(urls, cred.ConnectionUrl)
+	}
+
+	imageID := ExtractIdFromLink(database.Links.DatabaseImage.GetHref())
+	if imageID == 0 {
+		return fmt.Errorf("Could not find database image ID")
+	}
+	serviceID := ExtractIdFromLink(database.Links.Service.GetHref())
+	if serviceID == 0 {
+		return fmt.Errorf("Could not find database service ID")
+	}
+	accountID := ExtractIdFromLink(database.Links.Account.GetHref())
+	if accountID == 0 {
+		return fmt.Errorf("Could not find database account ID")
+	}
+
+	image, _, err := client.ImagesAPI.GetDatabaseImage(ctx, imageID).Execute()
+	if err != nil {
+		return generateErrorFromClientError(err)
```

Comment:
New client errors can be used as-is

```suggestion
		return err
```